sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -367,7 +367,7 @@ case class UnresolvedFunction(
     arguments: Seq[Expression],
     isDistinct: Boolean,
     filter: Option[Expression] = None,
-    ignoreNulls: Boolean = false,
+    ignoreNulls: Option[Boolean] = None,
```

Comment:
One question is if we really want to encourage every function to define its own default null behavior when IGNORE/RESPECT NULLS is not specified. `collect_list`/`collect_set` functions ignore nulls by default, which can be a legacy exception.

cc @srielau

Reply:
How many are there?

Reply:
It's only `collect_list/collect_set`.